tap_github/repository_streams.py

```diff
@@ -1307,3 +1307,55 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class StatsContributorsStream(GitHubStream):
+    """
+    Defines 'StatsContributors' stream. Fetching contributors activity.
+    https://docs.github.com/en/rest/reference/metrics#get-all-contributor-commit-activity
+    """
+
+    name = "stats_contributors"
+    path = "/repos/{org}/{repo}/stats/contributors"
+    primary_keys = ["org", "repo", "user_id"]
+    parent_stream_type = RepositoryStream
+    ignore_parent_replication_key = False
+    state_partitioning_keys = ["repo", "org"]
+    # Note - these queries are expensive and the API might return an HTTP 202 if the response
+    # has not been cached recently. https://docs.github.com/en/rest/reference/metrics#a-word-about-caching
+    tolerated_http_errors = [202]
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of flattened contributor activity."""
+        parsed_response = super().parse_response(response)
+        for contributor_activity in parsed_response:
+            weekly_data = contributor_activity["weeks"]
+            for week in weekly_data:
+                # no need to save weeks with no contributions.
+                is_week_empty = sum(week[key] for key in ["a", "c", "d"]) > 0
+                if is_week_empty:
+                    continue
+                week_with_author = dict()
+                week_with_author.update(week)
+                week_with_author.update(contributor_activity["author"])
+                yield week_with_author
+
+    schema = th.PropertiesList(
+        # Parent keys
+        th.Property("repo", th.StringType),
+        th.Property("org", th.StringType),
+        th.Property("user_id", th.IntegerType),
+        # Activity keys
+        th.Property("w", th.IntegerType),
```

Comment:
I'd be tempted to rename these fields so they're more or less self-documenting:
- `week_start`
- `additions`
- `deletions`
- `commits`

What do you think?